traffic_ops/traffic_ops_golang/user/user.go

```diff
@@ -217,16 +221,45 @@ func (user *TOUser) Read() ([]interface{}, error, error, int) {
 			return nil, nil, fmt.Errorf("parsing user rows: %v", err), http.StatusInternalServerError
 		}
 
+		// role is a required field for the endpoint, but not for an item in the database
+		// I doubt a nil check is needed, but I'm including it just in case
+		if user.RoleName == nil {
+			return nil, nil, fmt.Errorf("role name is nil", err), http.StatusInternalServerError
+		}
+		user.RoleNameGET = user.RoleName
+		user.RoleName = nil
+
 		users = append(users, *user)
 	}
 
 	return users, nil, nil, http.StatusOK
 }
 
+func (user *TOUser) privCheck() (error, error, int) {
+	privLevel, _, err := dbhelpers.GetPrivLevelFromRoleID(user.ReqInfo.Tx, *user.Role)
```

Comment:
Not sure how that will make it clearer since I read `requestedPrivLevel` as a getter. `privCheck` makes sure the requested priv level is accessible to the current user. I'll add a comment.

Reply:
Oh, I thought you meant change the method name.